File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        int[] r = new int[n];
        Stack<Integer> stack = new Stack<>();
        for(int i = 0 ; i < n ; i++){
            int x = temperatures[i];
            while(!stack.isEmpty() && x > temperatures[stack.peek()] ){
                int y = stack.pop();
                r[y] = i - y; 
            }
            stack.push(i);
        }
        return r;
    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = deque()
        ans = [0] * len(temperatures)
        for i, t in enumerate(temperatures):
            val = 1
            while len(stack) > 0 and stack[-1][0] < t:
                _, idx = stack.pop()
                ans[idx] = i - idx
            stack.append((t, i))                    
        return ans
            